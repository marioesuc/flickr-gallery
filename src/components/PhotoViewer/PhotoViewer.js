import React from 'react';
import './PhotoViewer.css';

// Custom PhotoViewer component
class PhotoViewer extends React.Component {
  handleOnArrowClick = relativeIndex => this.props.onArrowClick(relativeIndex);
  handleOnViewerClose = () => this.props.onViewerClose();

  render() {
    const { visible, photo } = this.props;

    if (visible) {
      const { source, title } = photo;

      return (
        <div>
          <div className='PhotoViewer-ImageContainer'>
            <img alt={source} className='PhotoViewer-image' src={source} />
            <div className='PhotoViewer-ArrowsContainer'>
              <div
                className='PhotoViewer-LeftArrowContainer'
                onClick={() => this.handleOnArrowClick(-1)}
              >
                ←
              </div>
              <div
                className='PhotoViewer-RightArrowContainer'
                onClick={() => this.handleOnArrowClick(1)}
              >
                →
              </div>
            </div>
            <div className='PhotoViewer-Title'>
              <span>{title}</span>
            </div>
          </div>
          <button
            className='PhotoViewer-close'
            onClick={this.handleOnViewerClose}
          >
            Close
          </button>
          <div
            className='PhotoViewer-background'
            onClick={this.handleOnViewerClose}
          />
        </div>
      );
    }

    return null;
  }
}

export default PhotoViewer;
