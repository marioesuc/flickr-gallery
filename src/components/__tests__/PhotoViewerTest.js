import React from 'react';
import { shallow, configure } from 'enzyme';
import PhotoViewer from '../PhotoViewer/PhotoViewer';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('PhotoViewer', () => {
  it('renders correctly when active', () => {
    const props = {
      visible: true,
      photo: {
        source: 'source',
        title: 'title'
      },
      onViewerClose: jest.fn()
    };

    const wrapper = shallow(<PhotoViewer {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when inactive', () => {
    const props = {
      visible: false,
      photo: null,
      onViewerClose: jest.fn()
    };

    const wrapper = shallow(<PhotoViewer {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
