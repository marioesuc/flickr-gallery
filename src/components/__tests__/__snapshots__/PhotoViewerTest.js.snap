// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoViewer renders correctly when active 1`] = `ShallowWrapper {}`;

exports[`PhotoViewer renders correctly when inactive 1`] = `ShallowWrapper {}`;
